app/src/test/webapp/app/shared/esbmsgsvc.test.js

```diff
@@ -0,0 +1,37 @@
+describe("EsbMessageService", function() {
+  var EsbMessageService;
+  var $httpBackend;
+
+  var resubmitRequest = {
+    "id": 1,
+    "payload": {
+      "body": "<foo>bar</foo>"
+    }
+  };
+
+  var resubmitResponse = {
+    "status": "Status"
+  };
+
+  beforeEach(module("esbMessageAdminApp"));
+
+  beforeEach(inject(function(_$httpBackend_, _EsbMessageService_){
+    $httpBackend = _$httpBackend_;
+    EsbMessageService = _EsbMessageService_;
+  }));
+
+  describe("after calling resubmit", function(){
+
+    it("should return a promise with the server response", function() {
```

Comment:
This test is good, but I care only partially about the response being returned. More interesting to me is that it calls the right API with the right parameters. You are testing that too, which is good, but the description doesn't tell me that. Can we reword this to include this more important information about the test?

For example, the full name of this test is "after calling resubmit should return a promise with the server response" but really it's more like "should POST resubmitted message payloads to api/key/resubmit/{id} and return response" ... see what I mean?

At a glance from the test name we should get an idea of the contract we're testing here. With a name like that, we know exactly what's expected of the API (and just as important: what we expect to be implemented in the production HTTP service).